### PR TITLE
allow private requirements in query groups

### PR DESCRIPTION
**Pull request by matklad**

### CODE REVIEW

File: tests/requires.rs [hunk 1]
```diff
@@ -0,0 +1,66 @@
+//! Test that transparent (uncached) queries work
```
Comment (nikomatsakis) [inline tests/requires.rs]: This comment looks out of date

File: tests/requires.rs [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+        &self.runtime
+    }
+}
+
+#[test]
+fn require_clauses_work() {
+    use queries::{InputGroup, PubGroup};
+    let mut db = Database::default();
+
+    db.set_input(1, 10);
```
Comment (nikomatsakis) [inline tests/requires.rs]: Of course, this isn't really testing that `db.private_a(x)` gets a compilation error -- but that's kind of hard to do right now without some form of "compiletest" like thing. I can't think of way to do that so I guess I'll take your word for it =)

Comment (matklad) [reply tests/requires.rs]: Wait a second, I guess we can use name clash for this? like, define `fn foo` for both private and public, and check that the public one is called.

Comment (matklad) [reply tests/requires.rs]: Ah, that won't work, because we'd get a clash of query storage structs...

**Pull request merged**